### DISCUSSION TITLE
Autogenerate rules.mk?

### DISCUSSION THREAD

**Discussion by NickLitten** [IBM/ibmi-tobi#334] [Q&A] [answered]
I have a bunch of source code (SQL, DDS, RPGLE, SQLRPGLE, SRVPGM's. etc) in a PC Folder and want to try out IBMi-BOB... so while I am sorting and organizing the folders to fit into a neat GIT repository... how can I auto-generate each folders Rules.MK file?
I know I will have to edit it afterwards to finesse various modules, rpgle, srvpgms, etc... but an autogenerate feature would be *excellent*

Any help for this #BobNoob?

**Comment by rbeckham** [comment, marked as answer]
@NickLitten look at using the package called SourceOrbit.  This will auto generate either Rules.mk files if using Bob or it can generate make files as well....also it will create a json object of dependencies if needed.  so you would use `so -bf bob` on the command line at the root of your project.

**Reply by NickLitten** [reply]
Thankyou!!!! That's exactly what I am looking for :)

**Comment by rbeckham** [comment]
You can run this on your local pc to generate the build files like you are asking for, and you can also use this in a Github Actions(Azure, etc) pipeline to automate the build of your application using DevOps.   I am trying to work on this now as well.   Lliam has also written another package called IBMi-CI  that you can use in your dev ops pipeline to automate the deployment of your application.

**Comment by NickLitten** [comment]
Update - this now easily done using the PROJECT EXPLORER | SOURCE ORBIT | BUILD BOB to rebuild the make files. :)
